File: BCW5/PHP/Base/Point.php
<?php
class Point {
        private $x, $y;

        private function validate($value) {
            if ( is_numeric($value) ) {
                return $value;
            }
            throw new Exception("Invalid param");
        }

        public function __construct($x=0, $y=0) {
            $this->x = $this->validate($x);
            $this->y = $this->validate($y);
        }

        public function __get($name) {
            return $this->$name;
        }

        public function __set($name, $value) {
            $this->$name = $this->validate($value);
        }

        public function distance(Point $other) {
            return hypot($this->x-$other->x, $this->y-$other->y);
        }

        public function __toString() {
            return sprintf("(%g, %g)", $this->x, $this->y);
        }
    }

   function compare($x, $y) {
        if ( $x == $y ) {
            echo $x . " is equal to " . $y . PHP_EOL;
        } else {
            echo $x . " is not equal to " . $y . PHP_EOL;
        }
    }


$object = new Point(1, "a");
$object1 = new Point(6, 0);

echo $object->distance($object1) . PHP_EOL;
var_dump($object);


?>

File: BC - w4/Car/Point.cpp
#include <iostream>
#include <cmath>
#include "point.h"
    Point::Point(double x, double y) {
        this->x = x;
        this->y = y;
    }
    Point::~Point(){}

    double Point::getX()const{
        return this->x;
    }
    double Point::getY()const{
        return this->y;
    }

    double Point::distance(const Point &b)const{
    return hypot(this->x-b.x, this->y-b.y);
}

bool Point::operator==(const Point& other)const{
    return this->x == other.x && this->y == other.y;
}

bool Point::operator!=(const Point& other)const{
    return !operator == (other);
}

std::ostream& operator << (std::ostream& out, const Point& point) {
    out << point.getX()<<" "<< point.getY() << std::endl;
    return out;
}